### ISSUE TITLE
Invalid hook call when installing recipes with NPM

### ISSUE THREAD

**Issue by dmvinson** [blitz-js/blitz#1953]
### What is the problem?
When running the `blitz install tailwind` command, the install fails with a "Invalid hook call" React error.

### Steps to Reproduce

1. Create new blitz project
2. Run `blitz install tailwind`

### Versions

```
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

debug: blitzPkgPath: /Users/me/projects/test/node_modules/blitz/dist/index.js
debug: cliPkgPath: /Users/me/projects/test/node_modules/@blitzjs/cli/lib/src/index.js

macOS Catalina | darwin-x64 | Node: v15.5.1

blitz: 0.30.0 (global)
blitz: 0.30.0 (local)

  Package manager: npm
  System:
    OS: macOS 10.15.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 1.01 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 15.5.1 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.3.0 - /usr/local/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: ~2.17 => 2.17.0
    blitz: 0.30.0 => 0.30.0
    prisma: ~2.17 => 2.17.0
    react: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    react-dom: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    typescript: 4.1.5 => 4.1.5

```

### Other

Please include applicable logs and screenshots that show your problem.
Command Output:
```
➜  test git:(master) blitz install tailwind
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose


  ERROR  Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
        1. You might have mismatching versions of React and the renderer (such as React DOM)
        2. You might be breaking the Rules of Hooks
        3. You might have more than one copy of React in the same app
        See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

 - 1. You might have mismatching versions of React and the renderer (such as React DOM)
 - 2. You might be breaking the Rules of Hooks
 - 3. You might have more than one copy of React in the same app
 - See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
 - resolveDispatcher (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/@blitzjs/installer/node_modules/react/cjs/react.development.js:1488:13)
 - Object.useReducer (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/@blitzjs/installer/node_modules/react/cjs/react.development.js:1528:20)
 - RecipeRenderer (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/@blitzjs/installer/src/recipe-renderer.tsx:157:35)
 - renderWithHooks (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:6036:18)
 - mountIndeterminateComponent (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:8570:13)
 - beginWork$1 (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:9938:16)
 - Object.invokeGuardedCallbackImpl (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:11563:10)
 - invokeGuardedCallback (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:11740:31)
 - beginWork$$1 (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:15778:7)
 - performUnitOfWork (/private/var/folders/9k/sxksxsx57hg0dflsbx4nyrxw0000gn/T/blitz-recipe-blitz-js-blitz/node_modules/ink/node_modules/react-reconciler/cjs/react-reconciler.development.js:14696:12)

```

**Comment by MrLeebo** [comment]
This appears to be caused when you are using NPM instead of Yarn. I can reproduce the error with the following steps:

```bash
blitz new repro
cd repro
rm yarn.lock # to trick blitz into thinking I don't have yarn
blitz install tailwind # doesn't matter which recipe you pick
```

There's probably a mistake in https://github.com/blitz-js/blitz/blob/canary/packages/installer/src/executors/add-dependency-executor.tsx when using NPM.

**Comment by MrLeebo** [comment]
> We are trying to call `npm add` instead of `npm install`. Should be pretty easy to fix.

This comment from @piotrski is right on the money I think.

**Comment by hirenchauhan2** [comment]
Hi, may I take this one and create a PR?

**Comment by flybayer** [comment]
@hirenchauhan2 yes!

**Comment by androsj** [comment]
still having the same issue on v0.36.0
npm: 7.6.0

**Comment by Rhinomcd** [comment]
I'm still having this same issue on windows. -- Opened a support thread in discord (#help-scorpius 3/24 ~7PM PST) and @Maastrich suggested to just switch to yarn.  (Thank you again btw)

That's fine for my usage, but I'm just commenting that I'm still able to reproduce this bug on windows in case anyone wants to dig into this further. 

blitz -v output:
```
rhino@RHINO-PC  ~\..\test-project   master ≢  blitz -v
You are using beta software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

Windows 10 | win32-x64 | Node: v15.10.0

blitz: 0.33.0 (global)
blitz: 0.33.0 (local)

  Package manager: npm
  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
    Memory: 35.46 GB / 63.90 GB
  Binaries:
    Node: 15.10.0 - ~\AppData\Roaming\npm\node.CMD
    Yarn: Not Found
    npm: 7.7.3 - ~\AppData\Roaming\npm\npm.CMD
    Watchman: Not Found
  npmPackages:
    @prisma/client: ~2.17 => 2.17.0
    blitz: 0.33.0 => 0.33.0
    prisma: ~2.17 => 2.17.0
    react: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    react-dom: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0
    typescript: ~4.1 => 4.1.5

**Comment by Rhinomcd** [comment]
Oops! I just realized this fix is slated for the 1.0 release.  My bad! I'm clearly new here ;)

**Comment by flybayer** [comment]
@Rhinomcd this is in the 1.0 milestone but closed because the mentioned issue was fixed.

Possible we have another issue, so if you can reproduce, please open a new issue.

But first make sure you have `legacy-peer-deps=true` in your `.npmrc` file
